packages/api/src/external/carequality/document/process-document-retrieval-results.ts

```diff
@@ -0,0 +1,170 @@
+import { out } from "@metriport/core/util/log";
+import { Bundle, BundleEntry } from "@medplum/fhirtypes";
+import { errorToString } from "@metriport/core/util/error/shared";
+import { capture } from "@metriport/core/util/notifications";
+import { DocumentReference } from "@metriport/ihe-gateway-sdk";
+import { MedicalDataSource } from "@metriport/core/external/index";
+import { DocumentRetrievalResult } from "../document-retrieval-result";
+import { isConvertible } from "../../fhir-converter/converter";
+import { setDocQueryProgressWithSource } from "../../hie/set-doc-query-progress-with-source";
+import { tallyDocQueryProgressWithSource } from "../../hie/tally-doc-query-progress-with-source";
+import { convertCDAToFHIR } from "../../fhir-converter/converter";
+import { upsertDocumentsToFHIRServer } from "../../fhir/document/save-document-reference";
+import { cqToFHIR } from "../../fhir/document";
+import { getDocuments } from "../../fhir/document/get-documents";
+import { ingestIntoSearchEngine } from "../../aws/opensearch";
+
+export async function processDocumentRetrievalResults({
+  requestId,
+  patientId,
+  cxId,
+  documentRetrievalResults,
+}: {
+  requestId: string;
+  patientId: string;
+  cxId: string;
+  documentRetrievalResults: DocumentRetrievalResult[];
+}): Promise<void> {
+  try {
+    for (const result of documentRetrievalResults) {
+      const { operationOutcome } = result.data;
+      const issuesWithGateway = operationOutcome?.issue?.length ?? 0;
+      const successDocsCount = result.data.documentReference?.length ?? 0;
+
+      tallyDocQueryProgressWithSource({
+        patient: { id: patientId, cxId: cxId },
+        progress: {
+          successful: successDocsCount,
+          errors: issuesWithGateway,
+        },
+        type: "download",
+        requestId,
+        source: MedicalDataSource.CAREQUALITY,
+      });
+
+      await handleDocReferences(result.data.documentReference, requestId, patientId, cxId);
```

Comment:
Don't we want to keep processing the results of other external GWs if we fail to process one of them?

Reply:
Yea we forsure we would, am i missing something though because it seems like it should handle all the results